### ISSUE TITLE
Null datetype value causing NPE

### ISSUE THREAD

**Issue by deanchen** [rstudio/sparklyr#193]
Regression introduced here preventing dataframe with null datetypes from being processed: https://github.com/rstudio/sparklyr/commit/7418d50d438912db1b04f2226b0efa8eb4f3ae7e#diff-34ea1a4718c2b36eb05f6ffba87130a5R127

```
Observations: NA
Variables: 133
Error : java.lang.NullPointerException
        at sparklyr.Utils$$anonfun$2.apply(utils.scala:127)
        at sparklyr.Utils$$anonfun$2.apply(utils.scala:125)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
        at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
        at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
        at sparklyr.Utils$.collectImplForceString(utils.scala:125)
        at sparklyr.Utils$.collectImpl(utils.scala:173)
        at sparklyr.Utils$$anonfun$collect$1.apply(utils.scala:183)
        at sparklyr.Utils$$anonfun$collect$1.apply(utils.scala:183)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.immutable.Range.foreach(Range.scala:160)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at sparklyr.Utils$.collect(utils.scala:183)
        at sparklyr.Utils.collect(utils.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sparklyr.Handler.handleMethodCall(handler.scala:118)
        at sparklyr.Handler.channelRead0(handler.scala:63)
        at sparklyr.Handler.channelRead0(handler.scala:15)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111)
        at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by kevinushey** [comment]
Thanks for the bug report -- it looks like https://github.com/rstudio/sparklyr/commit/7418d50d438912db1b04f2226b0efa8eb4f3ae7e#diff-34ea1a4718c2b36eb05f6ffba87130a5R127 needs a null check.

**Comment by javierluraschi** [comment]
@deanchen thanks for catching this, fix applied to `0.3.12`.
